init-common/functions

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+qsvc() {
+    # Returns whether a service is enabled.
+    # Usage: qsvc <nameofservice>
+    # Must only be used after qubes-sysinit has started.
+    test -e /var/run/qubes-service/"$1"
+}
+
+under_systemd() {
+    pidof systemd >/dev/null 2>&1
+}
+
+possibly_run_save_script()
+{
+    ENCODED_SCRIPT=$(qubesdb-read /qubes-save-script)
+    if [ -z "$ENCODED_SCRIPT" ] ; then return ; fi
+    tmpfile=$( mktemp qubes-save-script.XXXXXXXXX )
+    echo $ENCODED_SCRIPT|perl -e 'use MIME::Base64 qw(decode_base64); local($/) = undef;print decode_base64(<STDIN>)' >"$tmpfile"
+    chmod 755 "$tmpfile"
+    Xorg -config /etc/X11/xorg-preload-apps.conf :0 &
+    while ! [ -S /tmp/.X11-unix/X0 ]; do sleep 0.5; done
+    DISPLAY=:0 su - user -c "$tmpfile"
+    ret=$?
+    killall Xorg
+    rm -f "$tmpfile"
+    return $ret
+}
```

Comment:
This function is called by init-common/prepare-dvm-*

init-common/prepare-dvm-systemd.sh is called by the qubes-dvm.service unit.  I have added a Wants= and After= qubes-gui-agent.service which will guarantee that the unit is never called before the GUI agent has started.

The problem is with init-common/prepare-dvm-systemd.sh.  This one is called in initscript qubes-core-appvm, but I have no clue whether the initscript will be called before or after the agent.

We have a fundamental difference in when the DVM preparation happens, between systemd and sysvinit.  With systemd, it happens at some point during boot which we have not very much control over.  In the initscript, it happens practically immediately after /rw and /home (or /home/volatile) are prepared.  See initscript qubes-core-appvm.

When SHOULD it happen, is my question?  What sorts of pre and postconditions should be met before we start the DVM preparation step?  What's the step even do, like, what's the point?  Is it documented anywhere?